### PR TITLE
[SPARK-7221] [Streaming] Expose the current processed file name of FileInputDStream to the users

### PR DESCRIPTION
**Pull request by animeshbaranawal**
@akhilthatipamula

Introduced a parameter called display_filename in StreamingContext which if true displays the filenames being processed otherwise the file names will not be displayed.

I affirm that the contribution is my original work and I license the work to the project under the project's open source license.

### CODE REVIEW

File: streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala [hunk 1]
```diff
@@ -400,6 +400,17 @@ class StreamingContext private[streaming] (
...
   }
 
+  /** Private variable to display the filenames being processesd */
+  private[streaming] var display_filename : Boolean = false
+
+  /**
+   * Function sets the display_filename parameter
+   * @param flag paramater to which display_filename will be set
+   */
+  def display_filename_enable(flag: Boolean){
```
Comment (jerryshao) [inline streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala]: I think Spark follows CamelCase style.

File: streaming/src/main/scala/org/apache/spark/streaming/dstream/FileInputDStream.scala [hunk 2]
```diff
@@ -253,13 +254,19 @@ class FileInputDStream[K, V, F <: NewInputFormat[K,V]](
   private def filesToRDD(files: Seq[String]): RDD[(K, V)] = {
     val fileRDDs = files.map(file =>{
       val rdd = serializableConfOpt.map(_.value) match {
-        case Some(config) => context.sparkContext.newAPIHadoopFile(
+        case Some(config) => {
+          if(flag){ println("PROCESSING FILE :"+file) }
```
Comment (jerryshao) [inline streaming/src/main/scala/org/apache/spark/streaming/dstream/FileInputDStream.scala]: I think the reason why I file a wish JIRA to expose file name is that user want to get the file name in the run-time, but here your implementation is just println the file name on the console, so for end-user, how to use this file name? I think your current implementation cannot well address the problem.

**Comment by animeshbaranawal** [comment]
I tested this code by running the example HdfsWordCount.scala. The filenames being processed were displayed at run-time.

**Comment by jerryshao** [comment]
Yes, it indeed displayed on the console, but how to get the filenames in program way?

**Comment by animeshbaranawal** [comment]
What does it mean to get the filenames in program way?

**Comment by jerryshao** [comment]
I mean the code can get both file name with the content of file in the run-time, so user can do something additional depends on the file name in the run-time.

**Comment by animeshbaranawal** [comment]
So you want to access filename of the file being processed?

**Comment by srowen** [comment]
Yes, this does not solve the problem, and some of the code style is wrong (var names for example). Close this PR please?

**Pull request closed without merging**